src/app/App.js

```diff
@@ -924,7 +940,7 @@ class App extends EventEmitter {
 			historyState.scrollTop = this.popstateScrollTop;
 			historyState.scrollLeft = this.popstateScrollLeft;
 		}
-		this.updateHistory_(title, redirectPath, nextScreen.beforeUpdateHistoryState(historyState), opt_replaceHistory);
```

Comment:
Actually, looks like that piece is what makes it work. What he did is not correct, but the response.responseUrl doesn't have the hash information: https://xhr.spec.whatwg.org/#the-responseurl-attribute

So, after a request, when we try to get the request.responseUrl to see if the server did a redirect, we lose the hash information. We could try to restore it in case the responseUrl has the same path of the request url (no redirect happened). What do you guys think?